packages/app/components/Settings/SettingsPage.tsx

```diff
@@ -45,13 +56,24 @@ export default function SettingsPage({
     error,
     mutate,
   } = useSWR(`api/v1/profile`, async () => API.profile.index());
+  const router = useRouter();
+  const { cid } = router.query;
+  const role = useRoleInCourse(Number(cid));
+  const isTAOrProfessor = role === Role.TA || role === Role.PROFESSOR;
 
   const [currentSettings, setCurrentSettings] = useState(
     defaultPage || SettingsOptions.PROFILE
   );
   const [uploading, setUploading] = useState(false);
+  const [isMobile, setMobile] = useState(false);
+  const windowWidth = useWindowWidth();
+  const [avatarSize, setAvatarSize] = useState(windowWidth / 2);
 
-  const avatarSize = useWindowWidth() / 10;
+  useEffect(() => {
+    setMobile(window.innerWidth < 768);
```

Comment:
since this is getting used in a few places, it might be worth abstracting the isMobile logic into a custom hook like
```
const useIsMobile = () => {
    const [isMobile, setIsMobile] = useState(false);
    useEffect(() => {
        setIsMobile(window.innerWidth < 768);
    });
   return isMobile;
}
```
I'm pretty sure this would work 🤔